### PR TITLE
Training Loop

### PR DESCRIPTION
**Pull request by andregustavo-ufpr**
This pull request introduces a new training script for an autoencoder model in `src/train.py`. The script demonstrates how to preprocess data, train an undercomplete autoencoder, and export the encoded results. The most important changes are as follows:

Model training and data preprocessing:

* Added data loading and preprocessing steps, including normalization with `StandardScaler` and conversion to PyTorch tensors.
* Implemented training logic for the `UndercompleteAE` autoencoder model, including loss calculation and optimizer setup.

Feature extraction and export:

* Encoded the input data using the trained model and exported the encoded features to a CSV file.

### CODE REVIEW

File: src/training/train.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    sys.exit(1)
+
+CSV_SOURCE = sys.argv[1]
+
+def setup():
+    """Sets up the environment by reading the CSV file and preparing the data."""
+
+    df = pd.read_csv(CSV_SOURCE)
+
+    df = df.drop('url', axis=1)
```
Comment (ferbcastro) [inline src/training/train.py:32]: consegue fazer salvar label junto com encoded data?

File: src/training/train.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    X = df
+
+    return X
+
+
+def train(X: pd.DataFrame):
+    """Trains the autoencoder on the provided data."""
+
+    # Normalizing Data
+    scaler = StandardScaler()
```
Comment (ferbcastro) [inline src/training/train.py:43]: acho que certo seria usar binary cross entropy mas nao garanto

**Comment by ferbcastro** [comment]
como modelo é exportado?

File: src/training/train.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        # Backward pass and optimization
+        optimizer.zero_grad()
+        loss.backward()
+        optimizer.step()
+
+        # Loss for each epoch
+        print(f'Epoch [{epoch + 1}/{num_epochs}], Loss: {loss.item():.4f}')
+
+
+    encoded_data = model.encoder(X_tensor).detach().numpy()
```
Comment (ferbcastro) [inline src/training/train.py:76]: nn entendi pq fazer encode do dados de treino

**Comment by ferbcastro** [comment]
perguntas
como modelo é exportado?

File: src/training/train.py (hunk 3 above)
Comment (andregustavo-ufpr) [reply src/training/train.py:76]: A ideia é usar o resultado do autoencoder como input do próximo modelo, e deixar o dado encodado diminui a dimensionalidade dele

File: src/training/train.py (hunk 2 above)
Comment (andregustavo-ufpr) [reply src/training/train.py:43]: Binary cross é pra erro, nao scaler, quer que troque do MSE?

File: src/training/train.py (hunk 3 above)
Comment (ferbcastro) [reply src/training/train.py:74]: isso sim, mas o modelo tem que ser validado antes

File: src/training/train.py (hunk 2 above)
Comment (ferbcastro) [reply src/training/train.py:41]: como entrada é um one hot encoding, não é necessário normalizar.
